pkg/providers/vsphere/vsphere.go

```diff
@@ -140,7 +142,18 @@ type IPValidator interface {
 	ValidateControlPlaneIPUniqueness(cluster *v1alpha1.Cluster) error
 }
 
-func NewProvider(datacenterConfig *v1alpha1.VSphereDatacenterConfig, clusterConfig *v1alpha1.Cluster, providerGovcClient ProviderGovcClient, providerKubectlClient ProviderKubectlClient, writer filewriter.FileWriter, ipValidator IPValidator, now types.NowFunc, skipIpCheck bool) *vsphereProvider { //nolint:revive
+// NewProvider initializes and returns a new vsphereProvider.
+func NewProvider(
+	datacenterConfig *v1alpha1.VSphereDatacenterConfig,
+	clusterConfig *v1alpha1.Cluster,
+	providerGovcClient ProviderGovcClient,
+	providerKubectlClient ProviderKubectlClient,
+	writer filewriter.FileWriter,
+	ipValidator IPValidator,
+	now types.NowFunc,
+	skipIPCheck bool,
```

Comment:
Can `skipIPCheck` eventually be refactored factored into the `skippedValidations` map?